zerver/lib/events.py

```diff
@@ -514,7 +514,7 @@ def apply_event(state: Dict[str, Any],
                         stream_data['subscribers'] = []
                     stream_data['stream_weekly_traffic'] = None
                     stream_data['is_old_stream'] = False
-                    stream_data['is_announcement_only'] = False
+                    stream_data['stream_post_policy'] = Stream.STREAM_POST_POLICY_EVERYONE
                     # Add stream to never_subscribed (if not invite_only)
                     state['never_subscribed'].append(stream_data)
```

Comment:
I think that we should actually be deleting the `is_announcement_only` line here -- I think it was a bug in the first place. 

Overall I'm very suspicious of this block of code as being wrong -- why aren't these values already present in the `send_event` call the leads to this?  Probably makes sense to fork this to a follow-up investigation since I don't want it to entangle this PR.  Opened as https://github.com/zulip/zulip/issues/13787; I think it's fine to merge this block as-is since it's not going to be a regression.  (And I'd like to avoid adding more to this PR so that we can merge a safe checkpoint on this work; it's already pretty huge)